cli/lsp/resolver.rs

```diff
@@ -83,12 +90,15 @@ struct LspScopeResolver {
 
 impl Default for LspScopeResolver {
   fn default() -> Self {
+    let factory = ResolverFactory::new(None);
```

Comment:
Below is mostly a refactor to use this new ResolverFactory struct which provides a more convient way of creating these services (FYI @nayeemrmn).